xilem_web/src/util.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use web_sys::wasm_bindgen::JsCast;
+
+/// Helper to get the HTML document body element
+pub fn document_body() -> web_sys::HtmlElement {
+    document().body().expect("HTML document missing body")
+}
+
+/// Helper to get the HTML document
+pub fn document() -> web_sys::Document {
+    let window = web_sys::window().expect("no global `window` exists");
+    window.document().expect("should have a document on window")
+}
+
+/// Helper to get a DOM element by id
+pub fn get_element_by_id(id: &str) -> web_sys::HtmlElement {
+    document()
+        .get_element_by_id(id)
+        .unwrap()
+        .dyn_into()
+        .unwrap()
+}
+
+/// Helper to get the value from an HTML input element from a given event.
+///
+/// Returns `None` if the event isn't a valid input event or conversions fail.
+pub fn event_target_value<T>(event: &T) -> Option<String>
+where
+    T: JsCast,
+{
+    event
+        .unchecked_ref::<web_sys::Event>()
+        .target()?
+        .dyn_into::<web_sys::HtmlInputElement>()
```

Comment:
I wonder, if that function should be called `event_target_value` when it's only valid for `HtmlInputElement` as target? AFAIK, there's quite a few other elements (e.g. `HtmlOptionElement`) having a `.value()`. I also don't think it needs to be generic over `T` as it's only used for `Deref<Target=web_sys::Event>`, so `&web_sys::Event` seems to be sufficient.

Reply:
> I wonder, if that function should be called `event_target_value` when it's only valid for `HtmlInputElement` as target?

You're right, i just took it over from leptos out of habit :see_no_evil: 

> AFAIK, there's quite a few other elements (e.g. `HtmlOptionElement`) having a `.value()`. 

:thinking: We can either create separate functions (as it is right now) or we can try out all possible elements within a generic function. Or is there another possibility?

> I also don't think it needs to be generic over `T` as it's only used for `Deref<Target=web_sys::Event>`, so `&web_sys::Event` seems to be sufficient.

:+1:

Reply:
I think we can keep it separate functions for now, less magic (which I think is good in this situation). In the future, e.g. with #440 we could filter events, in this case based on value.